libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/location/engine/GoogleLocationEngineImpl.java

```diff
@@ -73,9 +73,7 @@ public void requestLocationUpdates(@NonNull LocationEngineRequest request,
 
     @Override
     public void removeLocationUpdates(@NonNull LocationCallback listener) {
-        if (listener != null) {
```

Comment:
Removing this if statement causes the navigation to crash with this error
`Exception in onDidFinishLoadingStyle
                                                                                                    java.lang.NullPointerException: Listener must not be null
                                                                                                    	at com.google.android.gms.common.internal.Preconditions.checkNotNull(com.google.android.gms:play-services-basement@@18.1.0:2)
                                                                                                    	at com.google.android.gms.common.api.internal.ListenerHolders.createListenerKey(com.google.android.gms:play-services-base@@18.1.0:1)
                                                                                                    	at com.google.android.gms.internal.location.zzbp.removeLocationUpdates(com.google.android.gms:play-services-location@@21.0.1:6)
                                                                                                    	at com.mapbox.services.android.navigation.v5.location.engine.GoogleLocationEngineImpl.removeLocationUpdates(GoogleLocationEngineImpl.java:76)
                                                                                                    	at com.mapbox.services.android.navigation.v5.location.engine.GoogleLocationEngineImpl.removeLocationUpdates(GoogleLocationEngineImpl.java:32)
                                                                                                    	at com.mapbox.mapboxsdk.location.engine.LocationEngineProxy.removeLocationUpdates(LocationEngineProxy.java:49)
                                                                                                    	at com.ilgiroscopio.ecosuite.navigator.maplibre.EnhancedLocationEngine.removeLocationUpdates(EnhancedLocationEngine.kt:92)
                                                                                                    	at com.mapbox.mapboxsdk.location.LocationComponent.setLocationEngine(LocationComponent.java:775)
                                                                                                    	at com.ilgiroscopio.ecosuite.navigator.viewmodel.NavigationScreenViewModel.initializeNavigation$lambda$0(NavigationScreenViewModel.kt:155)
                                                                                                    	at com.ilgiroscopio.ecosuite.navigator.viewmodel.NavigationScreenViewModel.$r8$lambda$Al82-OxFehTMaCmmSkm-4JweKpk(Unknown Source:0)
                                                                                                    	at com.ilgiroscopio.ecosuite.navigator.viewmodel.NavigationScreenViewModel$$ExternalSyntheticLambda2.onStyleLoaded(Unknown Source:8)
                                                                                                    	at com.mapbox.mapboxsdk.maps.MapboxMap.notifyStyleLoaded(MapboxMap.java:962)
                                                                                                    	at com.mapbox.mapboxsdk.maps.MapboxMap.onFinishLoadingStyle(MapboxMap.java:224)
                                                                                                    	at com.mapbox.mapboxsdk.maps.MapView$MapCallback.onDidFinishLoadingStyle(MapView.java:1346)
                                                                                                    	at com.mapbox.mapboxsdk.maps.MapChangeReceiver.onDidFinishLoadingStyle(MapChangeReceiver.java:198)
                                                                                                    	at com.mapbox.mapboxsdk.maps.NativeMapView.onDidFinishLoadingStyle(NativeMapView.java:1124)
                                                                                                    	at android.os.MessageQueue.nativePollOnce(Native Method)
                                                                                                    	at android.os.MessageQueue.next(MessageQueue.java:335)
                                                                                                    	at android.os.Looper.loop(Looper.java:206)
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:8653)
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method)
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:602)
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1130)`

Reply:
Thanks, I am not sure about your code, maybe the `@NonNull` is wrong then. Is this happening with the sample app?

Reply:
Is you your app calling `removeLocationUpdates `? If yes, you can't pass a null value and your IDE/lint should also tell you about it. 

If not, it would be good to better understand what is happening. No need to share the code, but maybe you can debug the situation. The null check and the @NonNull together are kind of wrong.

The @NonNull is coming from `public interface LocationEngine` which is outside of this library. So I can't removed that here easily.

Reply:
In order to keep backwards compatibility I added the null check again and added a comment about this. I guess it's not worth to hunt bugs for such a small thing :)